app/ui/src/main/java/com/fsck/k9/activity/compose/ComposeCryptoStatus.kt

```diff
@@ -0,0 +1,162 @@
+package com.fsck.k9.activity.compose
+
+
+import com.fsck.k9.activity.compose.RecipientMvpView.CryptoSpecialModeDisplayType
+import com.fsck.k9.activity.compose.RecipientMvpView.CryptoStatusDisplayType
+import com.fsck.k9.activity.compose.RecipientPresenter.CryptoMode
+import com.fsck.k9.message.AutocryptStatusInteractor
+import com.fsck.k9.message.AutocryptStatusInteractor.RecipientAutocryptStatus
+import com.fsck.k9.message.CryptoStatus
+import com.fsck.k9.view.RecipientSelectView.Recipient
+import org.openintents.openpgp.OpenPgpApiManager
+import org.openintents.openpgp.OpenPgpApiManager.OpenPgpProviderState
+
+/** This is an immutable object which contains all relevant metadata entered
+ * during email composition to apply cryptographic operations before sending
+ * or saving as draft.
+ */
+data class ComposeCryptoStatus(private val openPgpProviderState: OpenPgpProviderState,
```

Comment:
More of a general comment: If there's a lot of code in a data class maybe that's a sign that the logic should be extracted to a separate class and only the output that is ready for consumption should be put into a data class.